File: src/main/java/com/bdev/hogwarts_api/connection/ConnectionWebListener.java
package com.bdev.hogwarts_api.connection;

import javax.servlet.ServletContextEvent;
import javax.servlet.ServletContextListener;
import javax.servlet.annotation.WebListener;

import static com.bdev.hogwarts_api.properties.PropertiesConfig.config;

@WebListener
public class ConnectionWebListener implements ServletContextListener {
    private SshConnection connection;

    @Override
    public void contextInitialized(ServletContextEvent servletContextEvent) {
        if (config.getDatabaseRemoteConnectionEnabled()) {
            connection = new SshConnection();
        }
    }

    @Override
    public void contextDestroyed(ServletContextEvent servletContextEvent) {
        if (config.getDatabaseRemoteConnectionEnabled()) {
            connection.close();
        }
    }
}

File: src/main/java/com/bdev/hogwarts_api/connection/ConnectionWebListener.kt
package com.bdev.hogwarts_api.connection

import javax.servlet.ServletContextEvent
import javax.servlet.ServletContextListener
import javax.servlet.annotation.WebListener

import com.bdev.hogwarts_api.properties.PropertiesConfig.config

@WebListener
open class ConnectionWebListener : ServletContextListener {
    private lateinit var connection: SshConnection

    override fun contextInitialized(servletContextEvent: ServletContextEvent) {
        if (config.databaseRemoteConnectionEnabled) {
            try {
                connection = SshConnection()
            } catch (e: Exception) {
                /* Do nothing */
            }
        }
    }

    override fun contextDestroyed(servletContextEvent: ServletContextEvent) {
        if (config.databaseRemoteConnectionEnabled) {
            connection.close()
        }
    }
}